zebedee/client.go

```diff
@@ -66,6 +67,19 @@ func New(zebedeeURL string) *Client {
 	}
 }
 
+// NewWithSetTimeoutAndMaxRetry creates a new Zebedee Client, with a configurable timeout and maximum number of retries
+func NewWithSetTimeoutAndMaxRetry(zebedeeURL string, timeout, retry int) *Client {
```

Comment:
This is perfectly fine and will solve the problem we were discussing. However, I would personally do something like this instead:

```
func NewWithSetTimeoutAndMaxRetry(zebedeeURL string, clienter dphttp.Clienter) *Client {
        hcClient := healthcheck.NewClientWithClienter(service, zebedeeURL, clienter)

	return &Client{
		hcClient,
	}
}
```
Accepting the dphttp clienter as a param removes the internal set up of the client from the scope of this function - you give it the client you want it to use pre configured (retries and timeout values are set by the code passing it into this func. It should decouple it from the config and allow you to test it easier.

That would be my approach, I wont enforce that though  I'll leave it up to you to decided.